### PR TITLE
Add mingw to GitHub CI

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -57,6 +57,20 @@ jobs:
...
+    name: Windows (win32) using mingw
+    runs-on: ubuntu-latest
+    container:
+      image: openrct2/openrct2-build:0.2.4-mingw
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Build OpenRCT2
+        run: |
+          sudo su
```
Comment (tupaschoal) [inline .github/workflows/ci.yml]: Why does it need `sudo`?

Comment (IntelOrca) [reply .github/workflows/ci.yml]: Ugh, well the container switches to a new user and that has a different uid / gid to the github actions checkout directory so you get permission issues. My quick fix was to change back to root.

Unfortunately the image we build in does not build anymore due to its third party dependencies changing or not existing so I can't make changes to it. This ideally needs to be resolved.

Comment (tupaschoal) [reply .github/workflows/ci.yml]: I see, that's unfortunate, thanks for the explanation :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -57,6 +57,20 @@ jobs:
...
               echo 'Not going to push build'
           fi
+  windows-mingw:
+    name: Windows (win32) using mingw
+    runs-on: ubuntu-latest
+    container:
+      image: openrct2/openrct2-build:0.2.4-mingw
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
```
Comment (Margen67) [inline .github/workflows/ci.yml]: v2 should be used.

Comment (AaronVanGeffen) [reply .github/workflows/ci.yml]: I had to look it up, so fwiw: https://github.com/actions/checkout#checkout-v2

**Pull request merged**